cuda_core/tests/test_program.py

```diff
@@ -12,6 +12,83 @@
 
 is_culink_backend = _linker._decide_nvjitlink_or_driver()
 
+def get_nvvm_ir():
+    """Generate working NVVM IR with proper version metadata"""
+    try:
+        from cuda.core.experimental._program import _get_nvvm_module
+        nvvm = _get_nvvm_module()
+        major, minor, debug_major, debug_minor = nvvm.ir_version()
```

Comment:
1. Could you leave a comment on why we need to test the versioned IR? When will this block raise (and switch to test the unversioned one)?
2. We should make this function a pytest fixture
3. We should create a skip condition in case libNVVM is not installed